examples/lines_bars_and_markers/fill_between_demo.py

```diff
@@ -29,50 +34,116 @@
+#
+# Selectively filling horizontal regions
+# --------------------------------------
+# The parameter *where* allows to specify the x-ranges to fill. It's a boolean
+# array with the same size as *x*.
+#
+# Only x-ranges of contiguous *True* sequences are filled. As a result the
+# range between neighboring *True* and *False* values is never filled. This
+# often undesired when the data points should represent a contiguous quantity.
+# It is therefore recommended to set ``interpolate=True`` unless the
+# x-distance of the data points is fine enough so that the above effect is not
+# noticeable. Interpolation approximates the actual x position at which the
+# *where* condition will change and extends the filling up to there.
+
+x = np.array([0, 1, 2, 3])
+y1 = np.array([1, 1, 0, 0])
+y2 = np.array([0, 0, 1, 1])
+
+fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
+
+ax1.set_title('interpolation=False')
+ax1.plot(x, y1, 'o--')
+ax1.plot(x, y2, 'o--')
+ax1.fill_between(x, y1, y2, where=(y1 > y2), color='C0', alpha=0.3)
+ax1.fill_between(x, y1, y2, where=(y1 < y2), color='C1', alpha=0.3)
+
+ax2.set_title('interpolation=True')
+ax2.plot(x, y1, 'o--')
+ax2.plot(x, y2, 'o--')
+ax2.fill_between(x, y1, y2, where=(y1 > y2), color='C0', alpha=0.3,
+                 interpolate=True)
+ax2.fill_between(x, y1, y2, where=(y1 <= y2), color='C1', alpha=0.3,
+                 interpolate=True)
 
 ###############################################################################
-# This example illustrates a problem; because of the data
-# gridding, there are undesired unfilled triangles at the crossover
-# points.  A brute-force solution would be to interpolate all
-# arrays to a very fine grid before plotting.
+#
+# .. note::
+#
+#    Similar gaps will occur if *y1* or *y2* are masked arrays. Since missing
+#    values cannot be approximated, *interpolate* has no effect in this case.
+#    The gaps around masked values can only be reduced by adding more data
+#    points close to the masked values.
 
 
 ###############################################################################
-# Use transforms to create axes spans where a certain condition is satisfied:
+#
+# Selectively marking horizontal regions across the whole Axes
+# ------------------------------------------------------------
+# The same selection mechanism can be applied to fill the full vertical height
+# of the axes. To be independent of y-limits, we add a transform that
+# interprets the x-values in data coorindates and the y-values in axes
+# coordinates.
+#
+# The following example marks the regions in which the y-data are above a
+# given threshold.
 
 fig, ax = plt.subplots()
-y = np.sin(4 * np.pi * x)
+x = np.arange(0, 4 * np.pi, 0.01)
+y = np.sin(x)
 ax.plot(x, y, color='black')
 
-# use data coordinates for the x-axis and the axes coordinates for the y-axis
 import matplotlib.transforms as mtransforms
 trans = mtransforms.blended_transform_factory(ax.transData, ax.transAxes)
```

Comment:
this is just `ax.get_xaxis_transform()`, and then you can kill the import above and the reference to blended_transform_factory in the import list below.